plugin/src/main/java/com/denizenscript/denizen/objects/properties/entity/EntityCharged.java

```diff
@@ -1,135 +1,74 @@
 package com.denizenscript.denizen.objects.properties.entity;
 
 import com.denizenscript.denizen.objects.EntityTag;
-import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.objects.Mechanism;
 import com.denizenscript.denizencore.objects.core.ElementTag;
-import com.denizenscript.denizencore.objects.properties.Property;
-import com.denizenscript.denizencore.objects.properties.PropertyParser;
-import org.bukkit.entity.Entity;
-import org.bukkit.entity.Guardian;
-import org.bukkit.entity.Vex;
-import org.bukkit.entity.WitherSkull;
-
-public class EntityCharged implements Property {
-
-    public static boolean describes(ObjectTag object) {
-        if (!(object instanceof EntityTag)) {
-            return false;
-        }
-        Entity entity = ((EntityTag) object).getBukkitEntity();
-        return entity instanceof WitherSkull
-                || entity instanceof Vex
-                || entity instanceof Guardian;
+import org.bukkit.entity.*;
+
+public class EntityCharged extends EntityProperty<ElementTag> {
+
+    // <--[property]
+    // @object EntityTag
+    // @name charged
+    // @input ElementTag(Boolean)
+    // @description
+    // If the entity is wither_skull, controls whether the skull is charged. Charged skulls are blue.
+    // If the entity is a vex, controls whether the vex is charging. Charging vexes have red lines.
+    // This is a visual effect, and does not cause the vex to actually charge at anyone.
+    // If the entity is a guardian, controls whether the guardian's laser is active.
+    // Note that guardians also require a target to use their laser, see <@link command attack>.
+    // If the entity is a ghast, controls whether the ghast is charging. Charging ghasts have a red mouth and eyes.
+    // This is a visual effect, and does not cause the ghast to actually shoot a fireball at anyone.
+    // -->
+
+    public static boolean describes(EntityTag entity) {
+        return entity.getBukkitEntity() instanceof WitherSkull
+                || entity.getBukkitEntity() instanceof Vex
+                || entity.getBukkitEntity() instanceof Guardian
+                || entity.getBukkitEntity() instanceof Ghast;

```

Comment:
adding ghast here is not part of modernizing the property, it's a separate change with issues of its own, should be considered separately